### PR TITLE
Use the new OCS sharee API endpoint in the share dropdown

### PR DESCRIPTION
**Pull request by nickvergessen**
- [x] Depends on #18234 - OCS endpoint for sharee list
- [x] Depends on #18185 - Share dialog moving to the right side bar
- [ ] Filter existing sharees out of the list

@DeepDiver1975 @rullzer @blizzz

### CODE REVIEW

File: core/js/share.js [hunk 1]
```diff
@@ -507,11 +507,22 @@ OC.Share={
...
-				$.get(OC.filePath('core', 'ajax', 'share.php'), { fetch: 'getShareWith', search: search.term.trim(), limit: 200, itemShares: OC.Share.itemShares, itemType: itemType }, function(result) {
+				$.get(
+					OC.linkToOCS('apps/files_sharing/api/v1') + 'sharees',
+					{
+						format: 'json',
+						fetch: 'getShareWith',
+						search: search.term.trim(),
+						limit: 200,
+						// FIXME need to get the share ids
+						// existingShares: OC.Share.itemShares,
```
Comment (nickvergessen) [inline core/js/share.js]: @PVince81 @blizzz is the list of the share IDs anywhere available in JS?

Comment (PVince81) [inline core/js/share.js]: Not really. You probably need the ones from the current file ? You might be able to extract it from the share dialog somehow... (good luck)

File: core/js/share.js [hunk 2]
```diff
@@ -509,11 +509,22 @@ OC.Share={
...
 				var $loading = $('#dropdown .shareWithLoading');
 				$loading.removeClass('hidden');
-				$.get(OC.filePath('core', 'ajax', 'share.php'), { fetch: 'getShareWith', search: search.term.trim(), limit: 200, itemShares: OC.Share.itemShares, itemType: itemType }, function(result) {
+				$.get(
+					OC.linkToOCS('apps/files_sharing/api/v1') + 'sharees',
+					{
+						format: 'json',
+						fetch: 'getShareWith',
+						search: search.term.trim(),
+						limit: 200,
```
Comment (nickvergessen) [inline core/js/share.js]: perPage

**Comment by rullzer** [comment]
Replaced by #21349 using the backbone stuff.

**Pull request closed without merging**